Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp

```diff
@@ -32,6 +32,21 @@
 #include <wtf/Lock.h>
 #include <wtf/PrintStream.h>
 
+#ifdef USE_OPENCDM_BCM_NEXUS_SVP
+#include <b_secbuf.h>
+#include <gst_brcm_svp_meta.h>
+
+struct RPCSecureBufferInformation {
```

Comment:
Tagging @pwielders here again.

This structure is something that is passed like this to OpenCDM contrary to a chunk of memory that we pass on the non nexus case. It looks like OpenCDM has already mechanisms to handle SVP memory. Given the information we pass, it looks like it can even handle subsamples. Besides, this structure is specified nowhere. I guess it should be inside OpenCDM.

So I think having all this logic (that is apparently present already) inside OpenCDM and returning the output pointer would be the way to go. If OpenCDM does not change the pointer, then we do nothing. If OpenCDM returns a new pointer then we create a meta.

Having this would probably save us from calling decrypt from the subsamples code path and the non subsamples one, we would have to compute only the subsamples and pass then to OpenCDM. I guess it is a good moment to propose this if you are writing a new C API and dropping the C++ one.

And another point that would require changes in the nexus plugin would be that we should not create a meta, we should put that memory inside the buffer because it makes no sense (IMHO) to keep the buffer with the encrypted frames and provide the decrypted result in a meta. The logical thing would be creating a GstMemory and replace the current one the the buffer, of create a new GstBuffer with the new memory and the same metadata as the other one.